File: hello-clice/check.py
#!/usr/bin/env python3
import os
import sys

def check():
    # Check if file exists and contains "Hello clice"
    try:
        with open("/workspace/output.txt", "r") as f:
            content = f.read().strip()
            if content == "Hello clice":
                return True
    except:
        pass
    return False

if __name__ == "__main__":
    sys.exit(0 if check() else 1)

File: hello-clice/check.pl
#!/usr/bin/env perl
# ubuntu:22.04 (hello-clice's base image) has no perl installed, so this
# checker should fail loudly and clearly when exec'd - not silently, and
# not with a stack trace that looks like a clice bug rather than a missing
# interpreter.
print "This should never run - perl isn't in the image.\n";
exit(0);